### PR TITLE
Rename/move IceInternal and IceUtilInternal C# flles

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR renames namespace IceInternal => Ice.Internal and moves its files to the subdirectory src/Ice/Internal.

Same for IceUtilInternal => Ice.UtilInternal and src/Ice/UtilInternal.

The main difficulty was dealing with Exception (Ice.Exception vs System.Exception) and SocketException (Ice.SocketException vs System.Net.Sockets.SocketException).

### CODE REVIEW

**Review by externl — APPROVED** [review]
Looks good to me!

File: cpp/src/slice2cs/Gen.cpp [hunk 1]
```diff
@@ -404,8 +404,9 @@ Slice::CsVisitor::writeDispatch(const InterfaceDefPtr& p)
     _out << sp;
     _out << nl << "public override bool ice_isA(string s, " << getUnqualified("Ice.Current", ns) << " current = null)";
     _out << sb;
-    _out << nl
-         << "return global::System.Array.BinarySearch(_ids, s, IceUtilInternal.StringUtil.OrdinalStringComparer) >= 0;";
+    _out
+        << nl
+        << "return global::System.Array.BinarySearch(_ids, s, Ice.UtilInternal.StringUtil.OrdinalStringComparer) >= 0;";
```
Comment (pepone) [inline cpp/src/slice2cs/Gen.cpp:409]: Not related to the PR but why don't we use `StringComparer.Ordinal` https://learn.microsoft.com/en-us/dotnet/api/system.stringcomparer.ordinal?view=net-8.0

**Review by pepone — APPROVED** [review]
Looks good, did you consider getting rid of `src/Ice/UtilInternal/` and moving these definitions to `src/Ice/Internal`?

File: cpp/src/slice2cs/Gen.cpp (hunk 1 above)
Comment (bernardnormier) [reply cpp/src/slice2cs/Gen.cpp:409]: Yes, we should fix that.

**Comment by bernardnormier** [comment]
> Looks good, did you consider getting rid of src/Ice/UtilInternal/ and moving these definitions to src/Ice/Internal?

We should do that too. I wanted to keep the logic of this PR as simple as possible and no do too much.

**Pull request merged**